public/app/types/events.ts

```diff
@@ -0,0 +1,7 @@
+import TimeSeries from 'app/core/time_series2';
+import TableModel from 'app/core/table_model';
+import { eventFactory } from '@grafana/data';
+import { DashboardModel } from 'app/features/dashboard/state';
+
+export const rendered = eventFactory<Partial<TableModel | TimeSeries[]>>('render');
```

Comment:
rename to panelRenderEvent ? or panelEventRender  (or namespace it inside an object along all the PanelEvents). Should not be rendered as this is not something that happens after render but an event to initiate rendering

Reply:
Thanks for pointing this out. Since it's a panel event does that suggest it should belong in @grafana/ui as well? If so I suppose TableModel/TimeSeries would need to be moved to one of the external packages as well